packages/flutter_test/lib/src/buffer_matcher.dart

```diff
@@ -4,7 +4,7 @@
 
 import 'dart:typed_data';
 
-import 'package:test_api/src/expect/async_matcher.dart'; // ignore: implementation_imports
+import 'package:matcher/src/expect/async_matcher.dart'; // ignore: implementation_imports
```

Comment:
For my own understanding, why is this one not part of the public API of package:matcher? Could it be? Seems like we are using it in a lot of places...

Reply:
> For my own understanding, why is this one not part of the public API of package:matcher?

AsyncMatcher was originally a hack in `package:test`. It's not a very easy class to use, and it's behavior is contradictory to the rest of `package:matcher`. 

> Could it be?

Now that is has been moved into `package:matcher` it would be a little less strange to expose it. I'd still be a bit wary to do so, since we don't want to spend effort on things like improving the documentation and compatibility with AsyncMatcher.

I think a better use of time is to focus on the transition to `package:checks` where async checks are fully supported.